components/org.wso2.carbon.identity.api.user.push.device.handler.v1/src/main/resources/push-auth.yaml

```diff
@@ -0,0 +1,780 @@
+openapi: 3.0.0
+info:
+  version: "v1"
+  title: "WSO2 Identity Server Push-Based Authentication Rest API "
+  description: |
+    This document specifies  **RESTfulAPIs** for WSO2 **Identity Server - Push-based Authentication**.
+    It is written with [openapi:3.0.0 ](http://swagger.io/).
+  contact:
+    name: "WSO2"
+    url: "http://wso2.com/products/identity-server/"
+    email: "iam-dev@wso2.org"
+  license:
+    name: "Apache 2.0"
+    url: "http://www.apache.org/licenses/LICENSE-2.0.html"
+
+servers:
+  - url: https://{host}/t/{tenant_context}/api/users/v1
+
+security:
+  - OAuth2: [ ]
+  - BasicAuth: [ ]
+paths:
+  /me/push-auth/discovery-data:
+    get:
+      description: |
+        This API is used to retrieve data for the QR Code to trigger the push authentication add account flow.<br/>
+         <b>Permission required:</b>
+         * /permission/admin/login
+      summary: |
+        Retrieve data for the QR code.
+      responses:
+        200:
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DiscoveryDataDTO'
+        401:
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+        403:
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+
+        500:
+          description: Internal Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorDTO'
+
+      tags:
+        - me
+
+  /me/push-auth/devices:
+    post:
+      description: |
+        This API is used to receive device information from the mobile app and complete the add account flow.<br/>
```

Comment:
Is this description correct? Feels like the description should be related to device registration

Reply:
Updated